src/modules/AieraChat/index.tsx

```diff
@@ -173,6 +185,110 @@ export function AieraChat(): ReactElement {
         [chatId, updateSession]
     );
 
+    // Subscribe to title updates channel
+    useEffect(() => {
+        let isEffectActive = true;
+        let currentChannel: RealtimeChannel | undefined;
+
+        const handleTitleChannelSwitch = async () => {
+            // If we're switching to a chat with a title channel
+            if (titleChannelName && (!subscribedTitleChannel || subscribedTitleChannel.name !== titleChannelName)) {
+                // Store reference to the old channel
+                const oldChannel = subscribedTitleChannel;
+
+                // Clear the subscribed channel state immediately
+                setSubscribedTitleChannel(undefined);
+
+                // Detach from old channel if it exists
+                if (oldChannel) {
+                    log(`Detaching from old title channel: ${oldChannel.name}`);
+                    await unsubscribeFromChannel(oldChannel.name);
+                }
+
+                // Only proceed if effect is still active
+                if (!isEffectActive) {
+                    return;
```

Comment:
Also curious why this is needed and if using a ref for the `subscribedTitleChannel` would help avoid this.